bindings/torch/tinycudann/modules.py

```diff
@@ -55,6 +55,79 @@ def backward(ctx, doutput):
 
 		return None, input_grad, weight_grad, None
 
+# NOTE: Jianfei: this is a temporary solution, when only grid has backward_backward and networks do not.
+class _module_function_encoding(torch.autograd.Function):
+	@staticmethod
+	def forward(ctx, native_tcnn_module, input, params, loss_scale):
+		# If no output gradient is provided, no need to
+		# automatically materialize it as torch.zeros.
+		ctx.set_materialize_grads(False)
+
+		native_ctx, output = native_tcnn_module.fwd(input, params)
+		ctx.save_for_backward(input, params, output)
+		ctx.native_tcnn_module = native_tcnn_module
+		ctx.native_ctx = native_ctx
+		ctx.loss_scale = loss_scale
+
+		return output
+
+	@staticmethod
+	def backward(ctx, doutput):
+		if doutput is None:
+			return None, None, None, None
+
+		if not doutput.is_cuda:
+			print("TCNN WARNING: doutput must be a CUDA tensor, but isn't. This indicates suboptimal performance.")
+			doutput = doutput.cuda()
+
+		input, params, output = ctx.saved_tensors
+		input_grad, weight_grad = _module_function_encoding_backward.apply(ctx, doutput, input, params, output)
+
+		return None, input_grad, weight_grad, None
+
+class _module_function_encoding_backward(torch.autograd.Function):
+	@staticmethod
+	def forward(ctx, ctx_fwd, doutput, input, params, output):
+		ctx.ctx_fwd = ctx_fwd
+		ctx.save_for_backward(input, params, doutput)
+		with torch.no_grad():
+			scaled_grad = doutput * ctx_fwd.loss_scale
+			input_grad, weight_grad = ctx_fwd.native_tcnn_module.bwd(ctx_fwd.native_ctx, input, params, output, scaled_grad)
+			input_grad = None if input_grad is None else (input_grad / ctx_fwd.loss_scale)
+			weight_grad = None if weight_grad is None else (weight_grad / ctx_fwd.loss_scale)
+		return input_grad, weight_grad
+
+	@staticmethod
+	def backward(ctx, dinput_grad, dweight_grad):
+		# NOTE: currently support:
+		#		✓	d(dL_dinput)_d(dL_doutput)	doutput_grad
+		#		✓	d(dL_dinput)_d(params)		weight_grad
```

Comment:
Minor nitpick: tabs should only be used for indentation (i.e. in the beginning of lines), not for alignment.